src/shapes/fabricObject.class.js

```diff
@@ -0,0 +1,12 @@
+import {
+  InteractiveFabricObject
+} from "../mixins/object_interactivity.mixin";
+
+// TODO somehow we have to make a tree-shakeable import
+
+export { InteractiveFabricObject as FabricObject }
+
+(function (global) {
+  const fabric = global.fabric;
+  fabric.Object = InteractiveFabricObject;
+})(typeof exports !== 'undefined' ? exports : window);
```

Comment:
if we don't find a better way, this together with the dead code remover plugin for rollup can do the optional imports.

Reply:
I am currently for the several entry points approach
```ts
import { Canvas, Text } from 'fabric/node';
import { Canvas, Text } from 'fabric/static'; // or other name
import { Canvas, Text } from 'fabric'; // interactive by default
```

Reply:
well the point is that Rect and Circle have to extend either from Object or Interactive Object, so that is why i get confused on how i change that throug imports.
Yes we can have different entry points that create 3 bundles and people that use the bundle entirely can do that, or they can just import the single TS files as they see fit. But at that point, you have to provide 2 version of Rect/Circle/Text...?

Till i m not done converting i ll ignore it

Reply:
A headache indeed
So Mixins, all of them

Reply:
Yes but it stays complicated mixin or class is still a bit complicated.

When we have a full ts folder we can worry about thay